weblate/trans/forms.py

```diff
@@ -133,6 +133,40 @@ def to_python(self, value):
         return value
 
 
+class DateRangeInput(forms.TextInput):
+    """Input compatible with date range picker."""
+
+    template_name = "snippets/range-field.html"
+
+
+class DateRangeField(forms.CharField):
+    """Field for a date range input."""
+
+    def __init__(self, **kwargs) -> None:
+        if "widget" not in kwargs:
+            kwargs["widget"] = DateRangeInput
+        super().__init__(**kwargs)
+
+    def to_python(self, value):
+        start, end = value.split(" - ")
+        return {
+            "start_date": from_current_timezone(datetime.strptime(start, "%m/%d/%Y")),
+            "end_date": from_current_timezone(datetime.strptime(end, "%m/%d/%Y")),
+        }
+
+    def validate(self, value):
+        if not (
+            isinstance(value["start_date"], datetime)
+            and isinstance(value["end_date"], datetime)
+        ):
+            raise ValidationError(gettext("Invalid date!"))
```

Comment:
This should also do the validation that start is before end (currently done in ReportsForm.clean).

Reply:
Transferred the implementation of this `validation that start is before end` inside the `DateRangeField.validate()`. A test was also added.

Reply:
The invalid date path is still not covered by the tests.

Reply:
The invalid test will never pass to that path since the `to_python` is already doing validation. I removed it since it is redundant and kept only the start_date < end_date check in `validate`.